src/kademlia/dht_tracker.cpp

```diff
@@ -284,30 +280,37 @@ namespace libtorrent { namespace dht {
 	void dht_tracker::update_storage_node_ids()
 	{
 		std::vector<sha1_hash> ids;
-		ids.push_back(m_dht.nid());
-#if TORRENT_USE_IPV6
-		ids.push_back(m_dht6.nid());
-#endif
+		for (auto& n : m_nodes)
+			ids.push_back(n.second.dht.nid());
 		m_storage.update_node_ids(ids);
 	}
 
+	node* dht_tracker::get_node(node_id const& id, std::string const& family_name)
```

Comment:
`dht_tracker::get_node` is used for the case where a node needs to reach into another node's routing table to get nodes to satisfy a `want` parameter.